src/trans/gpu/internal/uvtvd_mod.F90

```diff
@@ -120,11 +115,11 @@ SUBROUTINE UVTVD(KF_UV,PU,PV,PVOR,PDIV)
 !*       1.2      COMPUTE VORTICITY AND DIVERGENCE.
 
 #ifdef OMPGPU
-!$OMP TARGET TEAMS DISTRIBUTE PARALLEL DO COLLAPSE(3) PRIVATE(IR,II,IN,KM,ZKM,JN,ZJN) DEFAULT(NONE) &
-!$OMP& SHARED(D_NUMP,R_NTMAX,KF_UV,D_MYMS,PVOR,PV,PU,PDIV,ZEPSNM)
+!$OMP TARGET TEAMS DISTRIBUTE PARALLEL DO COLLAPSE(3) PRIVATE(IR,II,IN,KM,ZKM,ZJN) &
+!$OMP& SHARED(D,R,KF_UV,FG,PVOR,PV,PU,PDIV) DEFAULT(NONE)
 #endif
 #ifdef ACCGPU
-!$ACC PARALLEL LOOP COLLAPSE(3) PRIVATE(IR,II,IN,KM,ZKM,JN,ZJN) FIRSTPRIVATE(KF_UV) DEFAULT(NONE) &
+!$ACC PARALLEL LOOP COLLAPSE(3) PRIVATE(IR,II,IN,KM,ZKM,ZJN)  DEFAULT(NONE) &
```

Comment:
This breaks Cray ACC on AMD GPUs. Nvidia compiler sails past this.

Reply:
Thanks, I'll take a look on LUMI.

Reply:
Well spotted, and I'm not sure why I made this change. This PR is not supposed to change ACC behaviour. I'll revert this.